From b7a43dd07915679285e4c18af3a39bb93f980979 Mon Sep 17 00:00:00 2001
From: paolo <paolo@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 27 Jun 2014 14:33:39 +0000
Subject: [PATCH] /cp
 2014-06-27  Paolo Carlini  <paolo.carlini@oracle.com>

	PR c++/61614
	* semantics.c (finish_compound_literal): Revert r204228.

/testsuite
2014-06-27  Paolo Carlini  <paolo.carlini@oracle.com>

	PR c++/61614
	* g++.dg/ext/complit14.C: New.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_9-branch@212074 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/cp/ChangeLog                     |    5 +++++
 gcc/cp/semantics.c                   |    1 -
 gcc/testsuite/g++.dg/ext/complit14.C |   11 +++++++++++
 4 files changed, 21 insertions(+), 1 deletions(-)
 create mode 100644 gcc/testsuite/g++.dg/ext/complit14.C

diff --git a/gcc/cp/ChangeLog b/gcc/cp/ChangeLog
index 497f691..6df3351 100644
--- a/gcc/cp/ChangeLog
+++ b/gcc/cp/ChangeLog
@@ -1,3 +1,8 @@
+2014-06-27  Paolo Carlini  <paolo.carlini@oracle.com>
+
+	PR c++/61614
+	* semantics.c (finish_compound_literal): Revert r204228.
+
 2014-06-26  Adam Butcher  <adam@jessamine.co.uk>
 
 	PR c++/61537
diff --git a/gcc/cp/semantics.c b/gcc/cp/semantics.c
index 3619e27..dad7e28 100644
--- a/gcc/cp/semantics.c
+++ b/gcc/cp/semantics.c
@@ -2600,7 +2600,6 @@ finish_compound_literal (tree type, tree compound_literal,
   if ((!at_function_scope_p () || CP_TYPE_CONST_P (type))
       && TREE_CODE (type) == ARRAY_TYPE
       && !TYPE_HAS_NONTRIVIAL_DESTRUCTOR (type)
-      && !cp_unevaluated_operand
       && initializer_constant_valid_p (compound_literal, type))
     {
       tree decl = create_temporary_var (type);
diff --git a/gcc/testsuite/g++.dg/ext/complit14.C b/gcc/testsuite/g++.dg/ext/complit14.C
new file mode 100644
index 0000000..aed765d
--- /dev/null
+++ b/gcc/testsuite/g++.dg/ext/complit14.C
@@ -0,0 +1,11 @@
+// PR c++/61614
+// { dg-options "" }
+
+int Fn (...);
+
+void
+Test ()
+{
+  int j = Fn ((const int[]) { 0 });                    // OK
+  unsigned long sz = sizeof Fn ((const int[]) { 0 });  // Error
+}
-- 
1.7.1

